e2e/single-cluster/status_test.go

```diff
@@ -107,3 +117,175 @@ var _ = Describe("Checks status updates happen for a simple deployment", Ordered
+
+		// Errors about resources other than bundles or bundle deployments not being found at deletion time
+		// should be ignored, as they may result from other test suites.
+		Expect(out).ToNot(MatchRegexp(
+			`ERROR.*Reconciler error.*Bundle(Deployment)?.fleet.cattle.io \\".*\\" not found`,
+		))
+
+		// Deleting the targetNamespace is not necessary when the GitRepo did not successfully
+		// render, as in a few test cases here. If no targetNamespace was created, trying to delete
+		// the namespace will result in an error, which is why we are not checking for errors when
+		// deleting namespaces here.
+		_, _ = k.Delete("ns", targetNamespace)
+	})
+
+	expectNoError := func(g Gomega, conditions []genericcondition.GenericCondition) {
+		for _, condition := range conditions {
+			if condition.Type == string(fleet.Ready) {
+				g.Expect(condition.Status).To(Equal(corev1.ConditionTrue))
+				g.Expect(condition.Message).To(BeEmpty())
+				break
+			}
+		}
+	}
+
+	expectTargetingError := func(g Gomega, conditions []genericcondition.GenericCondition) {
+		found := false
+		for _, condition := range conditions {
+			if condition.Type == string(fleet.Ready) {
+				g.Expect(condition.Status).To(Equal(corev1.ConditionFalse))
+				g.Expect(condition.Message).To(ContainSubstring("Targeting error"))
+				g.Expect(condition.Message).To(
+					ContainSubstring(
+						"<.ClusterLabels.foo>: map has no entry for key \"foo\""))
+				found = true
+				break
+			}
+		}
+		g.Expect(found).To(BeTrue())
+	}
+
+	ensureClusterHasLabelFoo := func() (string, error) {
+		return k.Namespace("fleet-local").
+			Patch("cluster", "local", "--type", "json", "--patch",
+				`[{"op": "add", "path": "/metadata/labels/foo", "value": "bar"}]`)
+	}
+
+	ensureClusterHasNoLabelFoo := func() (string, error) {
+		return k.Namespace("fleet-local").
+			Patch("cluster", "local", "--type", "json", "--patch",
+				`[{"op": "remove", "path": "/metadata/labels/foo"}]`)
+	}
+
+	When("a git repository is created that contains a template error", func() {
+		BeforeEach(func() {
+			targetNamespace = testenv.NewNamespaceName("target", r)
+		})
+
+		It("should have an error in the bundle", func() {
+			_, _ = ensureClusterHasNoLabelFoo()
+			Eventually(func(g Gomega) {
+				status := getBundleStatus(g, k, gitrepoName+"-examples")
+				expectTargetingError(g, status.Conditions)
+			}).Should(Succeed())
+		})
+
+		It("should have an error in the gitrepo", func() {
+			_, _ = ensureClusterHasNoLabelFoo()
+			Eventually(func(g Gomega) {
+				status := getGitRepoStatus(g, k, gitrepoName)
+				expectTargetingError(g, status.Conditions)
+			}).Should(Succeed())
+		})
+	})
+
+	When("a git repository is created that contains no template error", func() {
+		It("should have no error in the bundle", func() {
+			_, _ = ensureClusterHasLabelFoo()
+			Eventually(func(g Gomega) {
+				status := getBundleStatus(g, k, gitrepoName+"-examples")
+				expectNoError(g, status.Conditions)
```

Comment:
Should we check the gitrepo status here as well, for the sake of completeness? (no strong opinions here)